### DISCUSSION TITLE
How to set MAX_BATTERY_CHARGE_CURRENT for multiple parallel battery banks

### DISCUSSION THREAD

**Discussion by herrep** [Louisvdw/dbus-serialbattery#831] [Q&A]
Hi,

My setup comprises two battery banks each controlled by an individual JKBMS. dbus-serialbattery shows both JKBMS in Cerbo GX. Clearly, only one BMS may control the maximum charge and discharge current.

Each individual battery bank is capable of handling 200 A charge and discharge current, while there is a system maximum of 500 A for charge and 210 A for discharge for the two battery banks altogether.

Therefore, I intended to set MAX_BATTERY_CHARGE_CURRENT to 400 A and MAX_BATTERY_DISCHARGE_CURRENT to 210 A, as these totals are still below the system limitation, while each battery bank is effected only by half of these values due to two parallel battery banks.

Unfortunately, this approach does not seem to work. When I look onto my Cerbo GX and go into the serialbattery details, the charge limitation is set to "BMS limit" which results in a maximum charge current of 200 A.

What did I miss to allow a total current of 400 A despite the individual BMS limitation of 200 A? Due to the parallel setup, even with a total current of 400 A only 200 A would arrive at each battery bank.

Best regards,
Peter

**Comment by mr-manuel** [comment]
Just read the FAQ :)

https://louisvdw.github.io/dbus-serialbattery/faq/#how-to-aggregate-multiple-batteries

**Reply by herrep** [reply]
Dear Manuel,

Thank you very much for your hint. I just wondered whether an aggregation service is mandatory to cope with my identified issue or if a modified configuration of dbus-serialbattery was sufficient.

After having briefly checked [dbus-aggregate-batteries](https://github.com/Dr-Gigavolt/dbus-aggregate-batteries), I came to the conclusion that this aggregation service is limited to aggregating batteries managed by a dbus-serialbattery compatible BMS ("This is a service for Victron Venus OS to collect the data from multiple parallel-connected batteries using https://github.com/Louisvdw/dbus-serialbattery driver, merge them and publish as a single virtual battery to Dbus. It could serve at least as a temporary solution for https://github.com/Louisvdw/dbus-serialbattery/issues/8").

However, my setup makes not only use of two JK BMS with dbus-serialbattery, but also has a CAN-BUS BMS. In view of these circumstances, I was not even sure whether the discussed aggregation services work at all.

Probably could someone shed some light onto this issue?

Thanks a lot,
Peter

**Reply by mr-manuel** [reply]
If you are not using an aggregator then you will not get the potential of your batteries.

Try the other aggregator. It it does not work you have to aggregate the data yourself with Node-RED and dbus-mqtt-battery.
